lib/Yield/PTML.cpp

```diff
@@ -195,6 +199,41 @@ static std::string tagTypeAsString(const yield::TagType::Values &Type) {
   }
 }
 
+using TagNode = BidirectionalNode<yield::Tag>;
+using TagTree = GenericGraph<TagNode>;
```

Comment:
> Is this an AST? Can it become an AST? If so, let's call it AST something something.

I guess it could qualify as a very basic AST, but I'm somewhat opposed to the terminology. (especially considering that I didn't even notice it was one until you explicitly pointed it out).

I'd rather keep it as a super-simple relation tree (better explanation in the comments is definitely due): for each tag we have, all the tags that are fully located inside of it (tags never intersect) are denoted as successors.

> Using graphs for this is an interesting idea that scares me a bit.

Actually, it was already a graph in my very first implementation, the one that got flattened when `html` got converted into `ptml`. Except I rolled my own basic tree data-structure at that point, since I wasn't as comfortable with using our `GenericGraph` then.

Reply:
To add to this, the only point of the graph is so when we iterate over the tags (in post order, one more advantage of the tree approach - we get the optimal order with a simple call as a bonus), for each one we can clearly see which tag it is located inside of (instead of having to iterate over all the previous tags to find the parent) and also the list of all the tags that are inside of it (letting us handle special cases on a node-per-node basis without impacting any unrelated tag in any way).

Reply:
There's an extended discussion on mattermost that eclipses this thread, so I'll close it.